ring-core/src/ring/util/response.clj

```diff
@@ -195,20 +216,36 @@
              dir-entry  (.getEntry jar-file (add-ending-slash entry-name))]
          (and dir-entry (.isDirectory dir-entry)))))
 
+(defmethod resource-info :jar
+    [^java.net.URL url]
+    (let [conn (.openConnection url)]
+      {:content        (delay (.getInputStream conn))
```

Comment:
Since we always use the :directory? value, and don't use the other values if it is false, we could do:

```clojure
(defmethod resource-info :jar
    [^java.net.URL url]
    (let [conn (.openConnection url)
          dir? (jar-directory? conn)]
      {:content        (if-not dir? (.getInputStream conn))
       :content-length (if-not dir?
                         (let [len (.getContentLength conn)]
                           (if (<= 0 len) len)))
       :directory?     dir?
       :last-modified  (if-not dir?
                         (let [last-mod (.getLastModified conn)]
                           (if-not (zero? last-mod)
                             (Date. last-mod))))}))
```